schedctl/calls.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	pb "github.com/bgzzz/go-schedule/proto"
+)
+
+// runRPC is wrapper for gRPC connection procedures
+// it connects to management node and runs specific call
+func runRPC(call func(client pb.SchedulerClient) error) error {
+
+	conn, err := grpc.Dial(Config.ServerAddress, grpc.WithInsecure())
+	if err != nil {
+		fmt.Printf("There is a problem with connection: %s\n", err.Error())
+		log.Error("did not connect: %v", err)
+		return err
+	}
+	defer conn.Close()
+	client := pb.NewSchedulerClient(conn)
+
+	err = call(client)
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+// ScheduleTasks receives command line arguments
+// for tasks schedule detective parses it in case of wrong
+// usage and calls the schedule gRPC and prints the result
+// to stdout
+func ScheduleTasks(args []string) error {
+
+	if len(args) < 2 {
+		errStr := "There is not enought arguments for tasks schedule cmd"
+		fmt.Println(errStr)
+		err := fmt.Errorf(errStr)
+		log.Error(err.Error())
+		return err
+	}
+
+	if !fileCheck(args[0]) {
+		err := fmt.Errorf("There is no pointer to tasks file (-f, --file)")
+		log.Error(err.Error())
+		return err
+	}
+
+	tl, err := ParseTasksFile(args[1])
+	if err != nil {
+		log.Error(err.Error())
+		return err
+	}
+
+	return runRPC(func(client pb.SchedulerClient) error {
```

Comment:
I'd just make a client and attach it to a struct and then you can use in it any method.